### DISCUSSION TITLE
timezones...arrrrrrggghhh

### DISCUSSION THREAD

**Discussion by oldwindmill** [amberelectric/public-api#28] [Q&A] [answered]
[response_1624449656795.json.txt](https://github.com/amberelectric/public-api/files/6701672/response_1624449656795.json.txt)
i'm having a moment.
its currently 22:00 in victoria, 23 jum. that's 12:00 Z if i recall...
just retrieved prices. the NEM time seems correct, but the start/end time seem to be 10 hours early. Shouldn't they be 12:00:01Z and 12:30:00Z?
 "type": "CurrentInterval",
    "date": "2021-06-23",
    "duration": 30,
    **"startTime": "2021-06-23T02:00:01Z",
    "endTime": "2021-06-23T02:30:00Z",
    "nemTime": "2021-06-23T22:30:00+10:00",**
    "perKwh": 36.52266,
    "renewables": 30.074,
    "spotPerKwh": 11.74596,
    "channelType": "general",
    "spikeStatus": "none",
    "tariffInformation": {
      "period": "peak"
    },
    "estimate": true
  },

russell
(attached the full response)

**Comment by madpilot** [comment]
Man, I hate timezones.

Yeah, that ain't right. I'll check it out in the morning.

**Reply by madpilot** [reply]
I think I worked it out. I'll blame myself, but it's way easier to blame JS

**Comment by madpilot** [comment, marked as answer]
This is fixed now.

**Reply by oldwindmill** [reply]
it is...Thanks

**Reply by edwapj** [reply]
It is also helpful to appreciate that nemtime is Australian Eastern Standard Time. 
So a program can run on AEST and always be compatible with nemtime .
This avoids the program having to deal with local time zones and daylight saving changeovers involving 23 and 25 hour days.

**Reply by madpilot** [reply]
Correct! One of the reasons I included the +10 in the time stamp, so parsing libraries do the right thing.
